File: sample-app/src/main/java/org/granchi/whatnow/framework/dagger/WhatNowComponent.java
package org.granchi.whatnow.framework.dagger;

import org.granchi.whatnow.WhatNowApplication;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Dagger component for the app.
 *
 * @author serandel
 */
@Singleton
@Component(modules = {WhatNowModule.class})
public interface WhatNowComponent {
    void inject(WhatNowApplication app);
}

File: sample-app/src/main/kotlin/org/granchi/whatnow/framework/dagger/WhatNowComponent.kt
package org.granchi.whatnow.framework.dagger

import dagger.Component
import org.granchi.whatnow.WhatNowApplication
import javax.inject.Singleton

/**
 * Dagger component for the app.
 */
@Singleton
@Component(modules = arrayOf(WhatNowModule::class))
interface WhatNowComponent {
    fun inject(app: WhatNowApplication)
}
